### ISSUE TITLE
Is it safe to remove RemainingEntities session global

### ISSUE THREAD

**Issue by phavekes** [OpenConext/OpenConext-engineblock#1619]
[This issue is imported from pivotal - Originaly created at Aug 13, 2019 by bstrooband](https://www.pivotaltracker.com/story/show/167870486)

When developing the SFO logic I encountered a RemainingEntities item in the Session superglobal. I\'ve tried to figure out if it was used somewhere and it seemed it wasn\'t used.

Could this be verified?

**Comment by phavekes** [comment]
@thijskh do you have any clue? (bstrooband - Aug 13, 2019)

**Comment by phavekes** [comment]
It seems it is used in `library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php` line 16 (Thijs Kinkhorst - Aug 14, 2019)

**Comment by phavekes** [comment]
_This is non blocking so don\'t feel any pressure to answer._

Let me clarify some things a bit more.

I had to refactor the code in order to add the SFO callout flow into EB but before consent and that is when I encountered the `$_SERVER[\'Processing\']` super global. As far as I know it was only used to store the initial state but I wasn\'t able to figure out the reason why it was fetched from the ProxyServer configuration. I did create a separate processing state handler to handle the consent and also handle the new SFO callout process. As far as I could see it all worked as expected and the tests were all green. I\'m however not 100% sure if all legacy code in Corto is tested enough to prove my assumption that it was only used for the consent flow.

I could not find anything other it was used for then to continue to the SP after consent but I wanted to validate it against your knowledge so I could be sure it really wasn\'t.


**Some things I encountered while diving into the code:**
The `$_SERVER[\'Processing\']` variable is initially set through the Corto adapter.
https://github.com/OpenConext/OpenConext-engineblock/blob/b3398de0c2bdf159652bfba1216506aaa4236388/library/EngineBlock/Corto/Adapter.php#L355

After that it is used to set the entities which still need to be processed:
https://github.com/OpenConext/OpenConext-engineblock/blob/b3398de0c2bdf159652bfba1216506aaa4236388/library/EngineBlock/Corto/Module/Service/AssertionConsumer.php#L98

And then there is a heck to see if the processing is done or there should be more processing.
https://github.com/OpenConext/OpenConext-engineblock/blob/b3398de0c2bdf159652bfba1216506aaa4236388/library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php#L40

I\'ve dived into the code to see why it was introduced and I was able to trace it back to 2011:
https://github.com/OpenConext/OpenConext-engineblock/blob/ab4e09187004dfd080fdba3222e7d8afb5c3b036/library/Corto/library/Corto/ProxyServer.php#L370


 (bstrooband - Aug 14, 2019)
